src/Schema.php

```diff
@@ -25,4 +25,28 @@ public static function deactivate() {
   public static function uninstall() {
   }
 
+  /**
+   * Cron event callback to ensure proper database indexes.
+   */
+  public static function cron_ensure_back_in_stock() {
+    global $wpdb;
+    $ids = $wpdb->get_col($wpdb->prepare("SELECT ID FROM {$wpdb->posts}
+      INNER JOIN {$wpdb->postmeta} pm ON {$wpdb->posts}.ID = pm.post_id
+      LEFT JOIN {$wpdb->postmeta} moeve ON {$wpdb->posts}.ID = moeve.post_id AND moeve.meta_key LIKE %s
+      WHERE pm.meta_key = %s
+        AND pm.meta_value <= CURRENT_DATE()
+        AND moeve.meta_id IS NULL", [
+          '_woocommerce-moeve_id_%',
+          '_' . Plugin::PREFIX . '_back_in_stock_date',
+    ]));
+
+    foreach ($ids as $id) {
+      delete_post_meta($id, '_' . Plugin::PREFIX . '_back_in_stock_date');
+      if (function_exists('wc_get_product')) {
+        $product = wc_get_product($product);
+        $product->save();
+      }
```

Comment:
1. Undefined variable `$product`.
1. I would have expected to only see the WooCommerce API here…? https://woocommerce.github.io/code-reference/classes/WC-Data.html#method_delete_meta_data
```suggestion
      if ($product = wc_get_product($id)) {
        $product->delete_meta_data('_' . Plugin::PREFIX . '_back_in_stock_date');
        $product->save();
      }
```

Reply:
If this runs on a project were we do not have WooCommerce available it will fail without the `function_exists` check, won't it?

Reply:
Good point, but then I would wrap the whole callback and perhaps even the cron event registration altogether into that condition.  Here in the middle of the loop / at the end of the action is a bit late. 😉